frontend/javascripts/oxalis/view/left-border-tabs/modals/add_segment_index_modal.tsx

```diff
@@ -0,0 +1,60 @@
+import { Modal } from "antd";
+import React, { useState } from "react";
+import { AsyncButton } from "components/async_clickables";
+import type { VolumeTracing } from "oxalis/store";
+import { api } from "oxalis/singletons";
+
+export default function AddSegmentIndexModal({
+  volumeLayerName,
+  volumeTracing,
+  hideAddSegmentIndexModal,
+}: {
+  volumeLayerName: string;
+  volumeTracing: VolumeTracing;
+  hideAddSegmentIndexModal: () => void;
+}) {
+  const [isCalculatingSegmentIndex, setIsCalculatingSegmentIndex] = useState(false);
+
+  const handleCalculateSegmentIndex = async () => {
+    setIsCalculatingSegmentIndex(true);
+    await api.tracing.addSegmentIndex(volumeTracing.tracingId);
+    setIsCalculatingSegmentIndex(false);
+  };
+
+  return (
+    <Modal
+      title="Add segment index to this layer"
+      onCancel={isCalculatingSegmentIndex ? undefined : hideAddSegmentIndexModal}
+      footer={null}
+      width={800}
+      maskClosable={false}
+      open
+    >
+      <p>
+        This will calculate the segment index for the volume annotation layer "{volumeLayerName}".
```

Comment:
feel free to add a sentence like "this enables features, such as computing segment statistics."